libwnbd/wnbd_ioctl.cpp

```diff
@@ -133,6 +137,245 @@ DWORD WnbdOpenAdapterCMDeviceInstance(PDEVINST DeviceInstance)
     return Status;
 }
 
+DWORD GetDrvDetailData(HDEVINFO DevHandle, PSP_DEVINFO_DATA DevInfoData,
+                       PSP_DRVINFO_DETAIL_DATA_A DrvDetailData)
+{
+    SP_DEVINSTALL_PARAMS_A InstallParams;
+    BOOL Cleanup = FALSE, Found = FALSE;
+    SP_DRVINFO_DATA_A DrvInfoData;
+    DrvInfoData.cbSize = sizeof DrvInfoData;
+    DWORD GLA = ERROR_SUCCESS;
+
+    InstallParams.cbSize = sizeof InstallParams;
+    if (!SetupDiGetDeviceInstallParamsA(DevHandle, DevInfoData, &InstallParams)) {
+        GLA = GetLastError();
```

Comment:
nit: let's rename GLA to Status so that we're consistent with the rest of the code. GLA also sounds a bit cryptic.